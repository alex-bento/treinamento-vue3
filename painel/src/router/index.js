import { createRouter, createWebHistory } from 'vue-router'

const Home = () => import('../views/home/index.vue')
const Feedbacks = () => import('../views/feedbacks/index.vue')
const Crendencial = () => import('../views/crendencial/index.vue')

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/crendencial',
    name: 'Crendencial',
    component: Crendencial,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

export default router
