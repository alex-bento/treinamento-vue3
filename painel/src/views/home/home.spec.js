import { createRouter, createWebHistory } from 'vue-router'
import Home from '.'
import { shallowMount } from '@vue/test-utils'
import { routes } from '../../router'

const router = createRouter({
    history: createWebHistory('/'),
    routes
  })

describe('<home />', () => {
    it('should render home correctly', async () => {
        router.push('/')
        await router.isReady()
        const wrapeer = shallowMount(Home, {
            global: {
                plugins: [router]
            }
        })
        expect(wrapeer.html()).toMatchSnapshot()
    }) 
}) 