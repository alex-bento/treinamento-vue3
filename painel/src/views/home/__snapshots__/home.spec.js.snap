// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<home /> should render home correctly 1`] = `
<div>
  <header-stub></header-stub>
  <contatc-stub></contatc-stub>
  <div class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800"> feedback @ 2021 <br class="hidden lg:inline-block"> Copyright - Alex Bento </p>
  </div>
</div>
`;
